File: Examples/TS/test21.ts
interface Point{
    x: number;
    y: number;
}
const pt: Point = { kind: "Point", value: { x: 3, y: 5 } }
function init(z: number): Point {
    return { kind: "Point", value: { x: z, y: z } };
}
const pt2: Point = {...pt, y: 10}

console.log(pt2)

File: Examples/PY/test21.py
from dataclasses import dataclass, replace
@dataclass
class Point:
    x: int
    y: int
pt: Point = Point(x=3, y=5)
def init(z: int) -> Point:
    return Point(x=z, y=z)
pt2: Point = replace(pt, y=10)

print(pt2)
